test/sanity/code-smell/update-bundled.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# (c) 2018, Ansible Project
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+import fnmatch
+import json
+import os
+import os.path
+import re
+import sys
+from distutils.version import LooseVersion
+
+import packaging.specifiers
+
+from ansible.module_utils.urls import open_url
+
+
+BUNDLED_RE = re.compile(b'\\b_BUNDLED_METADATA\\b')
+
+
+def get_bundled_libs(paths):
+    bundled_libs = set()
+    for filename in fnmatch.filter(paths, 'lib/ansible/compat/*/__init__.py'):
+        bundled_libs.add(filename)
+
+    bundled_libs.add('lib/ansible/module_utils/distro/__init__.py')
+    bundled_libs.add('lib/ansible/module_utils/six/__init__.py')
+    bundled_libs.add('lib/ansible/module_utils/compat/ipaddress.py')
+    # backports.ssl_match_hostname should be moved to its own file in the future
+    bundled_libs.add('lib/ansible/module_utils/urls.py')
+
+    return bundled_libs
+
+
+def get_files_with_bundled_metadata(paths):
+    with_metadata = set()
+    for path in paths:
+        if path == 'test/sanity/code-smell/update-bundled.py':
```

Comment:
Are we allowing bundled files to be placed anywhere in the source tree?

If not, then the test will be more efficient if the directories allowed are added to the .json file as prefixes. Assuming `test/sanity/code-smell/` isn't one of those directories then this conditional could also be removed.

Example of using prefixes:

https://github.com/ansible/ansible/blob/06d83bae056bbc9de904a238e480acf1147fe80e/test/sanity/code-smell/no-main-display.json#L5-L8

Reply:
Yeah, they're allowed anywhere.  I half heatedly tried a couple times to specify a single directory (or two directories, one for controller-only and one for module_utils) but it never got enough traction from others so there's no rule. If someone would care to make some rules and move the code around then the test could be changed later.